### ISSUE TITLE
False Positive for check "Keep single parameter on one line" for inline loops

### ISSUE THREAD

**Issue by an-so** [abaplint/abaplint#3512]
## Context
Check: [_Keep single parameter on one line_](https://rules.abaplint.org/keep_single_parameter_on_one_line/)

## Observed Behavior

The following code snippet leads to a finding. The inline loop is wrongly considered a parameter.

The problem is that the ABAP Cleaner formats the code like shown below which produces a finding.

```
result = some_method_call( VALUE #( FOR i = 2 UNTIL i > 4
                                    ( strings[ i ] ) ) ).
``` 

## Expected Behavior

* Inline loops should not be considered single parameters.
* Formatting rules should be consistent with the ABAP Cleaner

**Comment by larshp** [comment]
thanks

**Comment by larshp** [comment]
well, there is one parameter to `some_method_call`

> Formatting rules should be consistent with the ABAP Cleaner

no, thats not a goal of abaplint

**Comment by larshp** [comment]
closing, moving to https://github.com/SAP/styleguides/issues/359

**Comment by larshp** [comment]
as a workaround suggest doing

```
FINAL(bar) = VALUE #( FOR i = 2 UNTIL i > 4
                                    ( strings[ i ] ) ).
result = some_method_call( bar ).
```

then it will also be more visible in the debugger
